code/modules/admin/verbs/debug.dm

```diff
@@ -301,33 +301,65 @@ But you can call procs that are of type /mob/living/carbon/human/proc/ for that
 	else
 		alert("Invalid mob")
 
-/client/proc/cmd_admin_change_hivenumber(mob/living/carbon/Xenomorph/M in mob_list, var/hivenumber = XENO_HIVE_NORMAL)
+/client/proc/cmd_admin_change_hivenumber(mob/living/carbon/Xenomorph/X in mob_list)
 	set category = "Debug"
 	set name = "Change Hivenumber"
 
+	if(!check_rights(R_DEBUG|R_ADMIN))	
+		return
+
+	if(!istype(X))
+		to_chat(usr, "This can only be done to instances of type /mob/living/carbon/Xenomorph")
+		return
+
+	var/hivenumber_status = X.hivenumber
+	var/list/namelist = list("Normal","Corrupted","Alpha","Beta","Zeta")
+
+	var/newhive = input(src, "Select a hive.", null, null) in namelist
+
+	if(!X || !istype(X))
+		return
+
+	var/newhivenumber
+	switch(newhive)
+		if("Normal")
+			newhivenumber = XENO_HIVE_NORMAL
+		if("Corrupted")
+			newhivenumber = XENO_HIVE_CORRUPTED
+		if("Alpha")
+			newhivenumber = XENO_HIVE_ALPHA
+		if("Beta")
+			newhivenumber = XENO_HIVE_BETA
+		if("Zeta")
+			newhivenumber = XENO_HIVE_ZETA
+
+	if(X.hivenumber != hivenumber_status)
+		to_chat(usr, "Someone else changed this xeno while you were deciding")
+		return
+
 	if(!ticker)
 		alert("Wait until the game starts")
 		return
 
-	if(M.gc_destroyed)
+	if(X.gc_destroyed)
 		alert("That mob doesn't seem to exist, close the panel and try again.")
 		return
 
-	if(isXeno(M))
-		log_admin("[key_name(src)] changed hivenumber of [M] to [M.hivenumber].")
-		M.hivenumber = hivenumber
-		if(istype(M, /mob/living/carbon/Xenomorph/Larva))
-			var/mob/living/carbon/Xenomorph/Larva/L = M
-			L.update_icons() // larva renaming done differently
-		else
-			M.generate_name()
-		if(istype(M, /mob/living/carbon/Xenomorph/Queen))
-			update_living_queens()
-		to_chat(usr, "Hivenumber set to [M.hivenumber]")
-		feedback_add_details("admin_verb","CHHN") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
-		message_admins("\blue [key_name(src)] changed hivenumber of [M] to [M.hivenumber].", 1)
+	log_admin("[key_name(src)] changed hivenumber of [X] to [newhive].")
+	message_admins("<span class='boldnotice'>[key_name(src)] changed hivenumber of [X] to [newhive].</span>", 1)
+	X.hivenumber = newhivenumber
```

Comment:
Actually turn this into a `carbon/Xenomorph/proc/set_hive_number()` that takes care of icon/name etc.